src/amo/components/Search/SearchForm.js

```diff
@@ -0,0 +1,70 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { loadEntities } from 'core/actions';
+import { fetchAddon } from 'core/api';
+import { gettext as _ } from 'core/utils';
+
+import 'amo/css/SearchForm.scss';
+import 'amo/css/lib/buttons.scss';
+
+export class SearchFormBase extends React.Component {
+  static propTypes = {
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string.isRequired,
+  }
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  goToSearch(query) {
+    const { pathname } = this.props;
+    this.context.router.push(`${pathname}?q=${query}`);
+  }
+
+  handleSearch = () => {
+    this.goToSearch(this.refs.query.value);
+  }
+
+  handleKeyDown = (e) => {
+    if (e.key === 'Enter') {
```

Comment:
We'll want to check this for cross-platform compat. It's fine in the admin but for AMO we'll need to ensure wider support.  http://caniuse.com/#feat=keyboardevent-key